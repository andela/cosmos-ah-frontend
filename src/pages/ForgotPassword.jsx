import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Form } from 'semantic-ui-react';
import styled from 'styled-components';
import { ButtonComponent } from '../components/Button';
import AltLogo from '../components/AppLogo';
import appLogo from '../assets/images/logo.png';


const Center = styled.div`
  text-align: center;
`;

const ContainerStyle = styled.div`
  margin: 0 auto;
  margin-top: 10%;
  width: 40%;
  padding: 30px;
  @media screen and (max-width: 768px) {
    width: 90%;
  }
  @media (min-width: 770px) and (max-width: 1115px) {
    width: 70%;
  }
`;

const HeaderStyle = styled.div`
  margin-top: 100px;
  h1 {
    letter-spacing: 2px;
    color: #484949;
  }
  @media screen and (max-width: 768px) {
    h1 {
      font-size: 23px;
    }
  }
  @media (min-width: 770px) and (max-width: 1115px) {
    h1 {
      font-size: 20px;
    }
  }`;

const FormStyle = styled(Form)`
  margin: 0 auto;
  &&& {
    input {
      background-color: #E8E8E8 !important;
      border: none !important;
    }
  }
`;

const AlignRight = styled.p`
  text-align: right;
  font-size: 17px;
  padding: 5px 0;
  @media screen and (max-width: 768px) {
    font-size: 14px;
  }
`;

const ForgotPassword = () => (


  <ContainerStyle>
    <Center>
      <AltLogo logo={appLogo} />
    </Center>
    <HeaderStyle>
      <h1>Forgot Password</h1>
    </HeaderStyle>
    <h4>Enter Your Email Address And We Will Email You With Instructions</h4>
    <FormStyle data-testid='forgotPassword'>
      <Form.Input size='big' icon={{ name: 'envelope', color: 'blue' }} iconPosition='left' fluid placeholder='YOUR EMAIL ADDRESS' />
      <ButtonComponent type='submit' size='big' color='blue'>RESET PASSWORD</ButtonComponent>
    </FormStyle>
    <AlignRight><Link to='/login'>Back to Login</Link></AlignRight>
  </ContainerStyle>
);

export default ForgotPassword;
