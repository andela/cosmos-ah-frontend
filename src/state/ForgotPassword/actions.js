import { userForgotPassword } from '../../lib/forgotPassword';
import { FORGOT_PASSWORD_SUCCESS, FORGOT_PASSWORD_FAILURE } from './actionTypes';

export const forgotPasswordSuccess = userPassword => ({
  type: FORGOT_PASSWORD_SUCCESS,
  payload: userPassword
});
export const forgotPasswordFailure = userPasswordError => ({
  type: FORGOT_PASSWORD_FAILURE,
  payload: userPasswordError
});

export const forgotPasswordAction = () => async dispatch => {
  try {
    const forgotPassword = await userForgotPassword();
    dispatch(forgotPasswordSuccess(forgotPassword));
  } catch (error) {
    dispatch(forgotPasswordFailure(error.response.data));
  }
};
