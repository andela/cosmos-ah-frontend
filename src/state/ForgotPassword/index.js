import { initialState } from './state';
import reducer from './reducers';

export const forgotPassword = {
  initialState,
  reducer,
  stateKey: 'forgotPassword'
};
