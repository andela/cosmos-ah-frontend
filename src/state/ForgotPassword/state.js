export const initialState = {
  forgotPassword: {
    message: '',
    error: null
  }
};
