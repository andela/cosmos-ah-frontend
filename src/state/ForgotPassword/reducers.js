import { initialState } from './state';
import { FORGOT_PASSWORD_SUCCESS, FORGOT_PASSWORD_FAILURE } from './actionTypes';

export default (state = initialState, action) => {
  switch (action.type) {
    case FORGOT_PASSWORD_SUCCESS:
      return {
        ...state,
        ...state.forgotPassword,
        message: action.payload,
        error: null
      };
    case FORGOT_PASSWORD_FAILURE:
      return {
        ...state,
        ...state.forgotPassword,
        error: action.payload
      };
    default:
      return state;
  }
};
