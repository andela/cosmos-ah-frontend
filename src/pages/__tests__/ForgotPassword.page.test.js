import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from 'react-testing-library';
import ForgotPassword from '../ForgotPassword';

afterEach(cleanup);

test('<ForgotPassword /> component', () => {
  const forgotPasswordPage = render(<BrowserRouter><ForgotPassword /></BrowserRouter>);
  expect(forgotPasswordPage).toBeTruthy();
  expect(forgotPasswordPage.getByText('RESET PASSWORD').tagName).toBe('BUTTON');
  expect(forgotPasswordPage.getByText('Back to Login').tagName).toBe('A');
});

test('renders the Form', () => {
  const { getByPlaceholderText } = render(<BrowserRouter><ForgotPassword /></BrowserRouter>);
  expect(getByPlaceholderText('YOUR EMAIL ADDRESS')).toBeTruthy();
});

test('renders the Header tag', () => {
  const forgotPasswordPage = render(<BrowserRouter><ForgotPassword /></BrowserRouter>);
  const H1Tag = forgotPasswordPage.getByText('Forgot Password');
  const H4Tag = forgotPasswordPage.getByText('Enter Your Email Address And We Will Email You With Instructions');
  expect(H1Tag.textContent).toBe('Forgot Password');
  expect(H4Tag.textContent).toBe('Enter Your Email Address And We Will Email You With Instructions');
  
});

test('renders the Component', () => {
  const forgotPasswordPage = render(<BrowserRouter><ForgotPassword /></BrowserRouter>);
  expect(forgotPasswordPage).toMatchSnapshot();
});
