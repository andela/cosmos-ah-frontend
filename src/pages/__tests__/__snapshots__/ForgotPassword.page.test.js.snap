// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htpNat bjYdsH"
      >
        <div
          class="sc-bwzfXH gJIwTT"
        >
          <div>
            <a
              href="/"
            >
              <img
                width="150"
              />
            </a>
          </div>
        </div>
        <div
          class="sc-bxivhb kPKGIS"
        >
          <h1>
            Reset Password
          </h1>
        </div>
        <h4>
          Enter Your Email Address And We Will Email You With Instructions
        </h4>
        <form
          class="ui form sc-bdVaJa dVBOwF"
          data-testid="forgotPassword"
        >
          <div
            class="field"
          >
            <div
              class="ui big left icon input"
            >
              <input
                name="email"
                placeholder="YOUR EMAIL ADDRESS"
                type="email"
                value=""
              />
              <i
                aria-hidden="true"
                class="blue envelope icon"
              />
            </div>
          </div>
          <div>
            <button
              class="ui blue big fluid button"
              type="submit"
            >
              RESET PASSWORD
            </button>
          </div>
        </form>
        <p
          class="sc-ifAKCX IRRQR"
        >
          <a
            href="/login"
          >
            Back to Login
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htpNat bjYdsH"
    >
      <div
        class="sc-bwzfXH gJIwTT"
      >
        <div>
          <a
            href="/"
          >
            <img
              width="150"
            />
          </a>
        </div>
      </div>
      <div
        class="sc-bxivhb kPKGIS"
      >
        <h1>
          Reset Password
        </h1>
      </div>
      <h4>
        Enter Your Email Address And We Will Email You With Instructions
      </h4>
      <form
        class="ui form sc-bdVaJa dVBOwF"
        data-testid="forgotPassword"
      >
        <div
          class="field"
        >
          <div
            class="ui big left icon input"
          >
            <input
              name="email"
              placeholder="YOUR EMAIL ADDRESS"
              type="email"
              value=""
            />
            <i
              aria-hidden="true"
              class="blue envelope icon"
            />
          </div>
        </div>
        <div>
          <button
            class="ui blue big fluid button"
            type="submit"
          >
            RESET PASSWORD
          </button>
        </div>
      </form>
      <p
        class="sc-ifAKCX IRRQR"
      >
        <a
          href="/login"
        >
          Back to Login
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
